### PR TITLE
Create Mn.Class - WIP

### LINKED ISSUE #2534: Metal - Create Mn.Class
We should add Mn.Class with Object feature parity:

```diff
+Marionette.Class = Metal.Class.extend({
+  // destroy: function() {
+  //   this.triggerMethod('before:destroy');
+  //   this.triggerMethod('destroy');
+  //   this.stopListening();
+  // },
+
+  // Import the `triggerMethod` to trigger events with corresponding
+  // methods if the method exists
+  triggerMethod: Marionette.triggerMethod,
+
+  // Proxy `getOption` to enable getting options from this or this.options by name.
+  getOption: Marionette.proxyGetOption,
+
+  // Proxy `bindEntityEvents` to enable binding view's events from another entity.
+  bindEntityEvents: Marionette.proxyBindEntityEvents,
+
+  // Proxy `unbindEntityEvents` to enable unbinding view's events from another entity.
+  unbindEntityEvents: Marionette.proxyUnbindEntityEvents
+
+});
```

I'm not sure if we should repurpose the tests here...

 Mn.mergeOptions
 Attach options to this.options in constructor
 proxyRadioHandlers in constructor

### PR DESCRIPTION
**Pull request by paulfalgout**
Resolves #2534 

I updated the WIP here and temporarily cherry-picked @megawac 's metal-fix.

### CODE REVIEW

File: src/metal.js [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
+  unbindEntityEvents: Marionette.proxyUnbindEntityEvents
+});
 
 function classify(obj) {
   return Marionette.Class.extend(
-    _.extend(
+    _.defaults(
       {constructor: obj},
-      _.omit(obj.prototype, _.keys(Marionette.Class.prototype))
+      Metal.Class.prototype,
```
Comment (megawac) [inline src/metal.js]: BTW, this can be changed to `Marionette.Class` now as `default` iterates in keys

File: src/metal.js [hunk 2]
```diff
@@ -1,14 +1,49 @@
 // Missing Metal Items
-// 1. Marionette.Class parity
+// 1. Marionette.Error
 // 2. Marionette.Deprecate
 
-Marionette.Class = Metal.Class.extend({});
+Marionette.Class = Metal.Class.extend({
+  constructor: function(options) {
+    this.options = _.extend({}, _.result(this, 'options'), options);
```
Comment (megawac) [inline src/metal.js]: Are we keeping `getOption` around?

File: src/metal.js [hunk 3]
```diff
@@ -1,14 +1,49 @@
...
+    this.cid = _.uniqueId(this.cidPrefix);
+    this.initialize.apply(this, arguments);
+  },
+
+  cidPrefix: 'mnc',
+
+  destroy: function() {
+    this.triggerMethod('before:destroy');
+    this.triggerMethod('destroy');
+    Marionette.unproxyRadioHandlers.apply(this);
```
Comment (megawac) [inline src/metal.js]: Should this be on prototype too ?

Comment (paulfalgout) [inline src/metal.js]: The radio handler stuff?  maybe.  I haven't really investigated it yet.  I don't know that the radio stuff belongs in the Class at all TBH.

**Comment by paulfalgout** [comment]
It looks to me as if most of the issues here have to do with using `Backbone.View` where if there's no `destroy` it's cleaning up the view with `remove`, but now that we're classifying it _does_ have a destroy.

File: src/metal.js (hunk 3 above)
Comment (paulfalgout) [inline src/metal.js]: With `Metal` it'd be cool if the radio stuff was just a mixin.  I dunno that it needs to be available on every single Mn and Bb class there is. But that would require https://github.com/marionettejs/backbone-metal/issues/20  or you'd have to init the radio handler mixin yourself.

**Comment by jasonLaster** [comment]
oh interesting... do we want to be clever here?

**Comment by paulfalgout** [comment]
@marionettejs/marionette-core 
Well this is improving.  Still `classify` issues at various versions.
The tests that were failing for the `destroy` issues are not up to date and need some work in general.

The implementation needs tests of its own still, but it is getting closer.

**Comment by jasonLaster** [comment]
@paulfalgout where are we at here?

**Comment by jasonLaster** [comment]
closing as this is 3.x work

**Pull request closed without merging**